src/Dfe.PlanTech.DatabaseUpgrader/Scripts/2024/20241212_1500_DatafixForDateLastUpdated.sql

```diff
@@ -0,0 +1,70 @@
+ALTER TABLE dbo.submission DISABLE TRIGGER tr_submission
+GO
+
+UPDATE S
+-- when the submission most recently had a response created, or fallback on the submission creation time
+SET dateLastUpdated = ISNULL(S.dateCompleted, ISNULL(LASTRESPONSE.dateCreated, S.dateCreated))
+FROM dbo.submission S
+CROSS APPLY (
+    SELECT TOP 1
+        R.dateCreated
+    FROM [dbo].response R
+    WHERE
+        R.submissionId = S.id
+    ORDER BY R.dateCreated DESC
+) LastResponse
+WHERE
+    -- Fix the updates that happened at 2024-12-12T10:05:14.7700000
+    dateLastUpdated BETWEEN DATETIMEFROMPARTS(2024, 12, 12, 10, 5, 14, 0)
```

Comment:
Not needed for this PR, but this kind of thing might be a good usecase for DbUp's [variable substitution support](https://dbup.readthedocs.io/en/latest/more-info/variable-substitution/#:~:text=Variable%20Substitution%20%2D%20DbUp&text=Variables%20can%20only%20contain%20letters,it%20is%20within%20a%20comment.)..

Reply:
Ah sweet, didn't know about that, thanks